app/controllers/concerns/time_interval.rb

```diff
@@ -0,0 +1,17 @@
+module TimeInterval
```

Comment:
You could also have a `TimeInterval` class which would have start_time and end_time.
Then `build_week_boundaries` would be a static method that builds an instance of TimeInterval.
Then a client of this class would use it:

```ruby
# use a factory method
time_interval = TimeInterval.week(params[:date])

# or initialize with whatever time you want
time_interval = TimeInterval.new(Time.parse('2016-07-28 15:00 EST'), Time.current)

# old time_interval method could be now renamed to something like this:
time_interval.collect_steps(30)
```

`week_start` looks like a private method, doesn't it?

Everything up to you :-)